### PR TITLE
tests added for controllers

### PR DESCRIPTION
**Pull request by Alibaba2023**
# Changes:

- Added tests for UserController and PostController(index and show actions)

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
+    # @user = User.find(params[:user_id])
+    # @posts = @user.posts
   end
 
   def index
-    @user = User.find(params[:user_id])
-    @post = @user.posts.find(params[:post_id])
+    # @user = User.find(params[:user_id])
+    # @post = @user.posts.find(params[:post_id])
   end
```
Comment (mckent05) [inline app/controllers/posts_controller.rb]: - You have done well so far, but please remove all commented code from your codebase, it'd help make your code cleaner and easier to understand 👍 👍

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,9 +1,9 @@
 class UsersController < ApplicationController
   def index
-    @users = User.all
+    # @users = User.all
   end
 
   def show
-    @user = User.find(params[:user_id])
+    # @user = User.find(params[:user_id])
   end
```
Comment (mckent05) [inline app/controllers/users_controller.rb]: - You have done well so far, but please remove all commented code from your codebase, it'd help make your code cleaner and easier to understand 👍 👍

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @Alibaba2023, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job adding test for your controllers 👍 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by roshan-bajgain — APPROVED** [review]
Hi @Alibaba2023, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- Gitflow is followed. ✔️
- Good job on changing the requested ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**